enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,339 @@
+
+The enhancement seeks to address the version independence of extra manifests, by putting version dependent manifests in version specific folder in git and pointing the directory via **extraManifestPath** and disabling the fetching builtIn RAN CRs provided from the ztp container 
+
+
+## Motivation
+
+* **Currently** customer can put custom manifests in the git repo and point them via **extraManifestPath**. 
+* **Current behavior:** If customer only want to include custom CRs to build site specific CRs:
+  * **inclusionDefault** must be set to *exclude* which will exclude all the CRs from ztp-container
+  * pass the list of CR file names under **filter.include**, that will include listed CRs.
+  * but it does not allow to have the [same CR name](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L462-L464) for the custom CRs. 
+
+  * to be able to add custom CRs, customer needs to change the name of the custom CR.
+
+
+  ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+  * as a result same named CR coming from user directory doesn't take precedent, rather siteconfig issues an error complaining same named CR is not allowed.
+
+### Goals
+
+- Allow customer to include any custom CRs having same/different name than builtIn CRs residing in the container
+- Provide a better control on how to enable and disable builtIn CRs
+- Keep backward compatibility of the behavior corresponds to 4.13 version.
+
+### Non-Goals
+
+- TBD
+### User Stories
+
+- [EPIC LINK](https://issues.redhat.com/browse/CNF-7365) --> [User Story](https://issues.redhat.com/browse/CNF-8672)
+
+## Proposal
+
+* **Controllability:** The proposal is to introduce a parameter called  <span style="color:red">*extraManifests.builtinCRs*</span>. which can hold value "*include|exclude*", which provide the flexibility for the customer to include/exclude CRs that is provided via ztp-container. 
+
+  * In this way, customer have full control of what CRs goes into the siteconfig. Until 4.13 release, customer couldn't put same named CR to the CR provided via ztp-container in the *extraManifestPath*. 
+
+
+ ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      builtinCRs: include | exclude   <----- New field
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+* **Version Independence**: This change will allow customer to have version independence of extra-manifests having different versioned folder in their git repository. Customer can extract versioned extra-manifests and can push to their git repo. An example has shown below:
+
+
+```yaml
+siteconfig/
+├── kustomization.yaml
+├── version_4.12
+│   ├── kustomization.yaml
+│   ├── sno-extra-manifest/
+│   ├── cnfde9.yaml
```

Comment:
How would the SiteConfig would look like in this case? Can we still have `extraManifestPath: mypath/`?

Reply:
Does that mean it will partially break backwards compatibility? Advantage of having `extraManifestPath` was to allow any CRs on top whatever came in the container. with `builtinCRs` set to `exclude`...it seems like we are giving that up flexibility?

Reply:
New commit https://github.com/openshift-kni/kni-enhancements/commit/3dadf26972770907a83001b5feddd3236416f943 is added based on the design discussion happened in 06/09 (https://docs.google.com/document/d/1S4RVy6dRmOTuVtPngB8YM2bUp4CUED_370XZJKPgoPo/edit#). Please review @pixelsoccupied

Reply:
I believe the current proposal preserves backwards compatibility and gives an opt-in ability to benefit from the search paths/version independence. In the case of a SiteConfig with only extraManifestPath, the behavior will be identical to the prior releases. If the user switches to the new (recommended) searchPath they get the new behavior.

Reply:
phrasing changed based on the suggestion.